### DISCUSSION TITLE
Option between photo and live photo

### DISCUSSION THREAD

**Discussion by Alejandro18650** [Radiokot/photoprism-android-client#126] [Ideas]
When I'm using the app watching live photos, I cannot see the photo because when I click on it, it start to autoplay the live motion. I think we should add a button for showing only the photo or the live, as google photos do.

**Comment by Alejandro18650** [comment]
For example, in google photos looks like this ![Screenshot_20240210_182055_Photos.jpg](https://github.com/Radiokot/photoprism-android-client/assets/92676426/9db605a4-0355-4dcf-8cc3-c8214bf4e33c)

**Comment by Radiokot** [comment]
Is this file a real live photo, or just a short video labeled "Live" by PhotoPrism?

**Reply by Alejandro18650** [reply]
A real live photo. I mean, the only way to see the photo of a live photo is in the preview.  If you open the photo, the live version of the photo is played

**Reply by Radiokot** [reply]
If it is a real live photo, you can open it with Google Photos using "More"->"Open with"
![Screenshot_20240211_090539_PhotoPrism Gallery](https://github.com/Radiokot/photoprism-android-client/assets/5675681/3cda716e-60f6-41d0-b2b0-3ad1d91e4fba)

**Reply by Radiokot** [reply]
Could you please send me a sample file? I'll make it play like live photo rather than a looped video.

**Reply by Alejandro18650** [reply]
Yeah obviusly. I know that there are many ways to see only the image but I think they are so annoying for the user. I think it has to be easier. We can do it similar to other galeries like google photos that has a button for switching from live to the photo.

I attach a compresed photo with live motion and also a video of what happens when I open the photo in the app
[real live photo.zip](https://github.com/Radiokot/photoprism-android-client/files/14313533/real.live.photo.zip)
https://github.com/Radiokot/photoprism-android-client/assets/92676426/7772e8c8-fb06-47cc-917c-1aea1a2aa788

**Reply by Radiokot** [reply]
Thank you for the sample, I'll release a hotfix today or tomorrow – it will properly play as live photo.

**Reply by Radiokot** [reply]
Here you go: https://github.com/Radiokot/photoprism-android-client/releases/tag/1.24.1(38)

**Comment by rrednaxela** [comment]
Related to this topic,
I have the issue that (Google) live photos take a very long time to load in the app. In the photoprism web client live photos don't autoplay, and show a still image unless you click the play button (which then results in the same loading behavior - but only after you clik play)
Is there a possibility to support such a behavior in the app, too?
From what I see the long loading time is caused by photoprism transcoding all live photos from mp4 to avc when viewed the first time. I've researched this a bit and it seems it's not possible to disable it. The problem is that my hardware is rather limited so this can easily take 15 seconds for 1 live photo. On the other hand I also don't want to transcode them all upfront and waste the disk space.

**Reply by Radiokot** [reply]
Do you upload your live photos in a way that splits them into JPEG and MP4?

**Reply by rrednaxela** [reply]
No, I'm uploading them unmodified - I'm attaching an example. 
![PXL_20240905_193422682 MP](https://github.com/user-attachments/assets/5c3a31d2-5e3d-476b-ab1c-f971be132c43)

**Reply by Radiokot** [reply]
PhotoPrism does not provide the codec of the video part of live photos, so the app relies on assumptions. If the file is JPG, it is assumed that the video part is AVC. If the file is HEIC, it is assumed that the video part is HEVC (unreleased change). In your case, it is JPG but HEVC. 

Possible workarounds for you:
- Try to make Google Camera use AVC for the video part, search for something like "compatibility mode";
- Shoot in HEIF (HEIC);
- Maybe do not shoot live photos if you do not view the video part anyway.

**Reply by Radiokot** [reply]
You could also try this gallery clone package with unreleased changes, which includes the ability to disable live photo animation
https://send.bitwarden.com/#QQ_3nx0NfkGj-rIGATGn_A/zKnSFhU-VZcNNXnTQ2iaoQ

**Reply by rrednaxela** [reply]
- just tried the unreleased package. The new setting works like a charm. Thanks a lot!
- I also tried the "compatibility mode" setting in the Google Camera app. It's well hidden, but you can opt to use H.264/AVC instead of H.265/HEVC. With this, the live photos play without delay. Obviously this works only for new photos and not for existing ones.
- Saving as HEIF/HEIC does not seem to be possible with the google camera app
- I previously considered not using live photos altogether, but occasionally it's actually a nice feature.

For the TV situation, where you want to watch photos with the family, disabling live photos seems like the best option for me right now.

So do I understand right that photoprism kind of splits the jpeg and video into separate files? I was wondering if you could do the "magic" completely on the client side by requesting the original file from photoprsim in case of live photos?
In my case I have anyway configured photoprism with the option
> To view original images, enable Dynamic Previews, and configure Dynamic Size Limit and Static Size Limit to a small value like 720. When viewing images exceeding that limit, the original files will be displayed.

So I guess that means in my case first the original image including the AVC/HEVC video is downloaded, and then the video separate as AVC?

**Reply by Radiokot** [reply]
Not all devices can play HEVC. If the app downloads the file, it contains HEVC video part while the device can't play it, then it is a failure. PhotoPrism, on the other hand, can at least transcode videos to a playable format.
